### DISCUSSION TITLE
Does beartype have a dedicated documentation website?

### DISCUSSION THREAD

**Discussion by Yura52** [beartype/beartype#206] [Q&A] [answered]
The latest release notes mention some changes to the documentation sphinx theme, but I cannot find the website itself. Is it available?

**Comment by posita** [comment]
I keep threatening to try to propose something here (probably around mkdocs and github.io), but then end up getting distracted by inconsequential nonsense like keeping my day job and working through the sci-fi gems that @leycec adds to my reading list. But know this: Such aspirations remain solidly on my list of things about which to fantasize actually doing someday. I have no doubts this bold declaration instills the utmost confidence in all who read it.

**Comment by leycec** [comment]
> working through the sci-fi gems that @leycec adds to my reading list.

**...yes. Yes! YESSSSSSS!!** Let's begin with an ancient big bang and end with a muted cybernetic whimper:

* [*Neutron Star*](https://www.goodreads.com/book/show/100366.Neutron_Star), the greatest anthology of short stories by Golden Age of Scifi luminary Larry Niven that nobody ever read despite introducing the Ringworld universe. Larry Niven himself infamously referred to *Neutron Star* as "The Worst Mistake He Ever Made." Don't believe Larry's bald-faced lies (with emphasis on "bald," because INTPs just don't come balder). Read *Neutron Star* and prepare to be astounded as you confront a post-human future beyond redemption! Exclamation points now included.
* [*The Moon Is a Harsh Mistress*](https://www.goodreads.com/book/show/16690.The_Moon_Is_a_Harsh_Mistress?from_search=true&from_srp=true&qid=9mms6qn93h&rank=3), Heinlein's best (and probably least known) treatise on stellar colonization and the ideological complications thereof. Ignore the excremental xenophobic politics he liberally smears over everything. Come for the just rebellion of ~~American~~ "Moon" colonists raging against the corrupt liberalism of their unjust ~~British~~ "Earth" masters. Stay for the libertarianism. And let's be clear: I abhor libertarianism. But that just proves how adept Robert "The Master" Heinlein was at the height of his formidable prowess. Even I enjoyed my stay on the Moon. `</chuckles>`
* **_The Dosadi Experiment._** Oh. My. Godness. So much Dosadi. So much experiment. Frank Herbert's unheralded masterpiece – and *easily* the densest scifi novel I've ever read. And I read (pretty much) every scifi novel, so you just know this is dense. Single sentences of this book span more compellingly eccentric thoughts than multiple chapters of competing novels. No link, because it's best to go in knowing nothing. Don't even read the jacket cover! But there's a catch, because even freedom in scifi isn't free:
  * **This is the fourth entry in the ConSentiency universe.** The first three (i.e., *A Matter of Traces,* *The Tactful Saboteur*, and *Whipping Star*) are all varying degrees of balls awful. I make no apologies for those short stories and books; quite simply, they're bad. Actually, *The Tactful Saboteur* is decent, but altogether too short. *Whipping Star* is the opposite. Sadly, you have to read them to make any headway with *The Dosadi Experiment.* The main protagonist of the first three ConSentiency stories reprises his role as humanity's monkey wrench in *The Dosadi Experiment.* That's a hard sell for even battle-hardened Hugo and Nebula readers. I blame no one for jettisoning themselves out the nearest airlock at *Whipping Star.* Just know that your suffering comes with a down payment of future awesomeness, guaranteed.
* [*Murderbot Diaries, The*](https://www.goodreads.com/series/191900-the-murderbot-diaries). Okay. So, you know how *basically* every good scifi yarn is really just a conceptual conceit in drag that obsessively examines some mostly unexamined facet of our real physical world that makes us deeply uncomfortable as we shift about in our faux leather executive chairs with sweaty foreheads? That's why scifi exists. Through the filmy gauze of scifi, we safely examine unsafe ideas that the active censors of our minds would otherwise prematurely terminate "with extreme prejudice." Politics is the obvious example. Through the grimy lens of scifi, we can debate, dabble, and prod at otherwise extreme political ideologies prohibited in everyday discourse. And *Murderbot Diaries* does superficially delve into the anti-corp dystopia that rents so much of modern scifi's mindshare... but anti-corp dystopia really isn't the crux. **High-functioning autism is the crux.** That's right! The bad-ass sociopathic artificial intelligence colloquially referred to by friends, family members, and itself as merely "Murderbot" (it means well, it swears) serves as an admittedly gore-splattered window into the obsessive-compulsive computing engine that drives highly introverted, very comical Asperger's. But there's the same catch as with *The Dosadi Experiment*, because even freedom in scifi isn't free:
  * **Murderbot only really _gits gud_** – like jaw-droppingly eyebrow-raisingly gud – **with its third entry,** _Artificial Condition_. Technically, the second entry _All Systems Red_ did win the Nebula for 2017. But who cares about Nebulas, am I right? Nebula winners are always just the worst of the worst, because scifi authors think they know best. Scifi authors think wrong! So, you'll have to sit through a novella and a full-length that @leycec publicly disrespects to get to the most compelling arc that modern scifi has to give: _Artificial Condition_ through _Network Effect_, comprising four full-lengths.

This is why we list *Dosadi* and *Murderbot* last. They're more work. Nobody likes work. Especially @leycec, as evidenced by this issue.

Class is adjourned, gentlemen.

![it's happening](https://media.tenor.com/T591TKMV5BgAAAAC/art-all.gif)

**Comment by leycec** [comment]
*...wait.* What were we talking about again?

**Oh...** *right.* Documentation. (I sigh quietly.)

## Where Muh @beartype Docos, Yo?

Your dark suspicions are sadly correct, @Yura52. Technically, [@beartype *does* host official documentation at Read The Docs (RTD)](https://beartype.readthedocs.io). Pragmatically, said documentation is basically non-existent *because I have no time to do it.* (Building extensive reading lists for @posita that span the entire history of science fiction may not be helping.)

Obviously, this is unacceptable. [Our current `README.rst` documentation](https://github.com/beartype/beartype/blob/main/README.rst) desperately needs to be split into smaller interlinked reStructuredText (reST) documents hosted at Read The Docs (RTD). In theory, I should have time do this; in practice, something inevitably breaks in @beartype or someone requests I create something interesting (which then breaks shortly after the next release) in @beartype.

No one likes griping; I shall now gripe. With only an hour a day of average free time for open-source volunteerism, I'm always at a loggerhead: "Do I code things everybody *needs* or do I document things everybody *wants*?" We see where I'm going with this. Needs beat up wants. I end up coding many things and documenting nothing. And... here we are. My face is exhaling with world-weariness. :face_exhaling:

**Comment by leycec** [comment, marked as answer]
**Everything above is a bald-faced lie!** [Beartype documentation now officially lives at ReadTheDocs (RTD)](https://beartype.readthedocs.io). Lo, I say:

* **https://beartype.readthedocs.io.**

 Our `README.rst` file has been truncated to a shadow of its former self. So this is what it feels like when bears cry.

![omg](https://media.tenor.com/JC2gzaAISiEAAAAC/sweaty-sweat.gif)
